fennel/test_lib/execute_aggregation.py

```diff
@@ -60,6 +60,27 @@ def get_val(self):
         return self.count
 
 
+class CountUniqueState(AggState):
+    def __init__(self):
+        self.count = 0
+        self.counter = Counter()
+
+    def add_val_to_state(self, val):
+        self.counter[val] += 1
+        self.count = len(self.counter)
+        return self.count
+
+    def del_val_from_state(self, val):
+        self.counter[val] -= 1
+        if self.counter[val] == 0:
+            del self.counter[val]
+        self.count = len(self.counter)
+        return self.count
+
+    def get_val(self):
+        return self.count
```

Comment:
To ensure that people expect 'approx' to work, will it make sense to arbitrarily add/remove a random amount upto 1% or so? We just need to be careful that for the same counter, same answer (even if it's approx) is returned on each call.

Reply:
I belive that is unnecessary complexity. We are making them write approx=True to make them aware.